src/include/optimizer/rule_impls.h

```diff
@@ -38,6 +38,22 @@ class InnerJoinCommutativity : public Rule {
                  OptimizeContext *context) const override;
 };
 
+/**
+ * @brief (A join B) join C -> A join (B join C)
+ */
+
+class InnerJoinAssociativity : public Rule {
```

Comment:
I think we need symmetry rule to perform this to perform A join (B join C) -> (A join B) join C.

Reply:
I’ve thought about this as well. I think with associativity and commutativity rule, you can achieve this by a series of transformations. It’s just we need to do those many steps instead of directly get there with another rule. But I think we’re going to exhaust the search space anyway, right?

Reply:
I see. Let's just keep it as is.